pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -160,6 +166,47 @@ func makeCredentialInitializer(serviceAccountName, namespace string, kubeclient
 	}, volumes, nil
 }
 
+func makeWorkingDirScript(workingDirs []string) string {
+	seen := map[string]bool{}
+	script := ""
+	for _, wd := range workingDirs {
+		if wd != "" {
+			p := filepath.Clean(wd)
+			if rel, err := filepath.Rel(workspaceDir, p); err == nil && !strings.HasPrefix(rel, ".") && !seen[p] {
+				seen[p] = true
+				if script == "" {
+					script = fmt.Sprintf("mkdir -p %s", p)
+				} else {
+					script = fmt.Sprintf("%s %s", script, p)
+				}
+			}
+		}
+	}
+
+	return script
+}
+
+func makeWorkingDirInitializer(steps []corev1.Container) *corev1.Container {
+	workingDirs := make([]string, len(steps))
```

Comment:
We could simplify this by making `workingDirs` a map (`map[string]bool` or even `map[string]struct{}`). That way we wouldn't have to worry about duplicates and thus wouldn't have to create the `seen` map above :angel:

Reply:
Good catch.